hypertrace-trace-enricher-impl/src/main/java/org/hypertrace/traceenricher/enrichment/enrichers/resolver/backend/JdbcBackendResolver.java

```diff
@@ -40,23 +45,45 @@ public JdbcBackendResolver(FQNResolver fqnResolver) {
         return Optional.empty();
       }
 
-      final String revisedBackendUriStr = backendUriStr.replaceFirst("^jdbc:", "");
-      URI backendURI = URI.create(revisedBackendUriStr);
-      String path = backendURI.getPath();
-      String dbScheme = backendURI.getScheme();
-      backendUriStr = String.format("%s:%d", backendURI.getHost(), backendURI.getPort());
+      // Split the URL based on two slashes to extract the host and port values.
+      String[] parts = backendUriStr.split("//");
+      String dbType;
+      if (parts.length > 1) {
+        backendUriStr = JDBC_EVENT_PREFIX + "://" + parts[1];
+        dbType = getDbType(parts[0]);
+      } else {
+        dbType = JDBC_EVENT_PREFIX;
```

Comment:
Should we be using `getDbType` here as well? Looks like oracleDb has this pattern `jdbc:oracle:<drivertype>:@<database>`

Reply:
That case will be handled in the `if` block @kotharironak